plugins/history_plugin/history_plugin.cpp

```diff
@@ -438,6 +452,57 @@ namespace eosio {
         return result;
       }
 
+      read_write::prune_history_result read_write::prune_history( const read_write::prune_history_params& params ) {
+         auto& chain = history->chain_plug->chain();
+
+         EOS_ASSERT( chain.get_read_mode() == db_read_mode::IRREVERSIBLE, chain::plugin_exception, "attempting to call prune history while NOT in irreversible read mode" );
+
+         chainbase::database& db = const_cast<chainbase::database&>( chain.db() ); // Override read-only access to state DB (highly unrecommended practice!)
+
+         prune_history_result result{0, false};
+
+         auto start_time = fc::time_point::now();
+
+         const auto& action_idx = db.get_index<action_history_index, by_id>();
+         auto itr_action = action_idx.lower_bound(history->last_id_pruned);
+
+         while( itr_action != action_idx.end() ) {
+           if(itr_action->block_num <= params.height) {
```

Comment:
Since `id` is incrementing with `block_num`, you can break out of loop once `block_num` > `params.height`.